### PR TITLE
Raven db 13726

### PR DESCRIPTION
**Pull request by maximburyak**

### CODE REVIEW

File: src/Voron/Impl/Paging/AbstractPager.cs [hunk 1]
```diff
@@ -54,6 +55,7 @@ protected AbstractPager()
             CanPrefetch = new Lazy<bool>(CanPrefetchQuery);
         }
 
+        public static string StaticLocker = "AbstractPagerStaticLocker";
```
Comment (ayende) [inline src/Voron/Impl/Paging/AbstractPager.cs]:
```suggestion
        public readonly static object WorkingSetIncreaseLocker = new object();
```

File: src/Voron/Impl/Paging/AbstractPager.cs [hunk 2]
```diff
@@ -79,10 +81,16 @@ public void SetPagerState(PagerState newState)
                                 if (DoNotConsiderMemoryLockFailureAsCatastrophicError)
                                     continue; // okay, can skip this, then
 
-                                var sumOfAllocations = Bits.PowerOf2(newState.AllocationInfos.Sum(x => x.Size) * 2);
+                                Monitor.Enter(StaticLocker);
```
Comment (ayende) [inline src/Voron/Impl/Paging/AbstractPager.cs]: why not `lock`?
And call `Sodium.sodium_mlock` again here

Comment (ayende) [reply src/Voron/Impl/Paging/AbstractPager.cs]: Need comment to explain why you are locking

File: src/Voron/Impl/Paging/Windows32BitsMemoryMapPager.cs [hunk 3]
```diff
@@ -238,15 +238,26 @@ public override bool ShouldLockMemoryAtPagerLevel()
             return false;
         }
 
+        public static string Windows32BitsMMapPagerStaticLocker = "Windows32BitsMMapPagerStaticLocker";
```
Comment (ayende) [inline src/Voron/Impl/Paging/Windows32BitsMemoryMapPager.cs]: why not same locker?

**Comment by arekpalinski** [comment]
1. Is `RavenDB-13726 ` used in commits the valid issue number - https://issues.hibernatingrhinos.com/issue/RavenDB-13726?
2. There are 2 test failures

File: src/Voron/Impl/Paging/CryptoPager.cs [hunk 4]
```diff
@@ -70,8 +70,8 @@ public unsafe class EncryptionBuffer
         public static readonly UIntPtr HashSize = Sodium.crypto_generichash_bytes();
         public static readonly int HashSizeInt = (int)Sodium.crypto_generichash_bytes();
         public byte* Pointer;
-        public int Size;
-        public int? OriginalSize;
+        public long Size;
```
Comment (arekpalinski) [inline src/Voron/Impl/Paging/CryptoPager.cs:73]: When dealing with large data (like attachments) we can have single allocation which are more than 2GB. So we ran into `int` overflow issue, right?

Comment (ayende) [reply src/Voron/Impl/Paging/CryptoPager.cs:73]: yes

**Review by ayende — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Comment by maximburyak** [comment]
Closing this in favor of https://github.com/ravendb/ravendb/pull/9656

**Pull request closed without merging**